include/nnvm/node.h

```diff
@@ -80,6 +82,7 @@ struct NodeAttrs {
    *  For place holder variable, op == nullptr.
    */
   const Op *op{nullptr};
+  std::shared_ptr<Graph> g;
```

Comment:
put this to the end of the current NodeAttrs

Reply:
add comments, the field name g is a bit too generic. Need to comment on what is the semantics here, is it just an attribute so that we can use to to compose a high order AST, or something else